modules/cli/src/main/scala/scala/cli/exportCmd/MavenProjectDescriptor.scala

```diff
@@ -0,0 +1,316 @@
+  ): MavenProject = {
+
+    val scalaV = getScalaVersion(options)
+    def getScalaMajorPrefix =
+      scalaV match {
+        case s"2.12.${patch}" => "2.12"
+        case s"2.13.${patch}" => "2.13"
+        case s"3.$x.$y"       => "3"
+      }
+
+    def buildMavenDepModels(
+      mainDeps: ShadowingSeq[Positioned[AnyDependency]],
+      isCompileOnly: Boolean
+    ) =
+      mainDeps.toSeq.toList.map(_.value).map { dep =>
+        val org  = dep.organization
+        val name = dep.name
+        val ver  = dep.version
+        // TODO dep.userParams
+        // TODO dep.exclude
+        // TODO dep.attributes
+        val artNameWithPrefix = dep.nameAttributes match {
+          case NoAttributes           => name
+          case s: ScalaNameAttributes => s"${name}_$getScalaMajorPrefix"
+        }
+        val scope0 =
+          if (scope == Scope.Test) MavenScopes.Test
+          else if (isCompileOnly)
+            MavenScopes.Provided // maven seems to support either test or provided, not both
+          else MavenScopes.Main
+
+        MavenLibraryDependency(org, artNameWithPrefix, ver, scope0)
+      }
+
+    val depSettings = {
+      def toDependencies(
+        mainDeps: ShadowingSeq[Positioned[AnyDependency]],
+        testDeps: ShadowingSeq[Positioned[AnyDependency]],
+        isCompileOnly: Boolean
+      ): Seq[MavenLibraryDependency] = {
+        val scopePriorities       = List()
+        val mainDependenciesMaven = buildMavenDepModels(mainDeps, isCompileOnly)
+        val testDependenciesMaven = buildMavenDepModels(testDeps, isCompileOnly)
+        val resolvedDeps = (mainDependenciesMaven ++ testDependenciesMaven).groupBy(k =>
+          k.groupId + k.artifactId + k.version
+        ).map { (_, list) =>
+          val highestScope = MavenScopes.getHighestPriorityScope(list.map(_.scope))
+          list.head.copy(scope = highestScope)
+        }.toList
+
+        val scalaDep = if (!ProjectDescriptor.isPureJavaProject(options, sources)) {
+          val scalaDep = if scalaV.startsWith("3") then "scala3-library_3" else "scala-library"
+          val scalaCompilerDep =
+            if scalaV.startsWith("3") then "scala3-compiler_3" else "scala-compiler"
+          List(
+            MavenLibraryDependency("org.scala-lang", scalaDep, scalaV, MavenScopes.Main),
+            MavenLibraryDependency("org.scala-lang", scalaCompilerDep, scalaV, MavenScopes.Main)
+          )
+        }
+        else Nil
+
+        resolvedDeps ++ scalaDep
+      }
+
+      toDependencies(
+        options.classPathOptions.allExtraDependencies,
+        testOptions.classPathOptions.allExtraDependencies,
+        true
+      )
+    }
+
+    MavenProject(
+      dependencies = depSettings
+    )
+  }
+
+  private def getScalaVersion(options: BuildOptions): String =
+    options.scalaParams.toOption.flatten.map(_.scalaVersion).getOrElse(
+      ScalaCli.getDefaultScalaVersion // FIXME account for pure Java projects, where Scala version isn't defined
```

Comment:
should we cover the `FIXME` comment in this PR, or extract to an issue?

Reply:
This shouldn't be an issue now. In maven, it is now handled based on purJava variable.